src/hotspot/share/runtime/crac.cpp

```diff
@@ -521,8 +521,20 @@ bool CracRestoreParameters::read_from(int fd) {
     return false;
   }
 
-  char *contents = NEW_C_HEAP_ARRAY(char, st.st_size, mtInternal);
-  if (read(fd, contents, st.st_size) < 0) {
+  // crac_restore_finalize() may terminate the process if we run on (older) CPU where glibc string functions may crash.
+  // The flag is stored separately as all the code of this function below is difficult to implement without the string functions.
+  bool IgnoreCPUFeatures_local;
+  if (read(fd, &IgnoreCPUFeatures_local, sizeof(IgnoreCPUFeatures_local)) != sizeof(IgnoreCPUFeatures_local)) {
+    perror("read (ignoring restore parameters)");
+    return false;
+  }
+  if (!IgnoreCPUFeatures_local) {
+    VM_Version::crac_restore_finalize();
```

Comment:
Is this function idempotent? It's called from `VM_Crac::doit()` as well (and it should be since reading the shm is optional), but could calling it twice cause any trouble?

Reply:
It is but it was just a luck. At least a comment there is wrong. I will improve it, thanks.